test/extensions/filters/http/tap/tap_filter_integration_test.cc

```diff
@@ -39,17 +40,64 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, TapIntegrationTest,
                          testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
                          TestUtility::ipTestParamsToString);
 
+// Verify a static configuration with an any matcher, writing to a file per tap sink.
+TEST_P(TapIntegrationTest, StaticFilePerTap) {
+  const std::string filter_config =
+      R"EOF(
+name: envoy.filters.http.tap
+config:
+  common_config:
+    static_config:
+      match_config:
+        any_match: true
+      output_config:
+        sinks:
+          - file_per_tap:
+              path_prefix: {}
+)EOF";
+
+  const std::string path_prefix = TestEnvironment::temporaryPath("");
+  initializeFilter(fmt::format(filter_config, path_prefix));
+
+  // Initial request/response with tap.
+  codec_client_ = makeHttpConnection(makeClientConnection(lookupPort("http")));
+  const Http::TestHeaderMapImpl request_headers_tap{{":method", "GET"},
+                                                    {":path", "/"},
+                                                    {":scheme", "http"},
+                                                    {":authority", "host"},
+                                                    {"foo", "bar"}};
+  IntegrationStreamDecoderPtr decoder = codec_client_->makeHeaderOnlyRequest(request_headers_tap);
+  waitForNextUpstreamRequest();
+  const Http::TestHeaderMapImpl response_headers_no_tap{{":status", "200"}};
+  upstream_request_->encodeHeaders(response_headers_no_tap, true);
+  decoder->waitForEndStream();
+
+  codec_client_->close();
+  test_server_->waitForCounterGe("http.config_test.downstream_cx_destroy", 1);
+
+  // Find the written .pb file and verify it.
+  auto files = TestUtility::listFiles(path_prefix, false);
+  auto pb_file = std::find_if(files.begin(), files.end(),
+                              [](const std::string& s) { return absl::EndsWith(s, ".pb"); });
```

Comment:
HTTP stream IDs are random numbers because we use them as stable seeds for runtime random route calculations. This is why I'm finding the file. This seemed like the easiest way of dealing with this. LMK if you can think of something better.

Reply:
Maybe extract from log? I guess it would be nice to decouple the entropy for runtime from the stream numbering, but that's a totally different PR.

Reply:
OK, can I think about changing this in a follow up? This works OK for now and isn't hard to understand IMO.